### PR TITLE
Update EFL and Enlightenment

### PR DESCRIPTION
**Pull request by ts468**
EFL: 1.12.2 -> 1.12.3
Enlightenment: 0.19.2 -> 0.19.3

### CODE REVIEW

**Comment by ts468** [comment]
builds fine locally

**Comment by vbgl** [comment]
Everything builds fine locally (linux x86_64 with chroot).

However it seems to me that the elementary library is not properly installed:

`Elementary.h`, line 66, reads: `#include "Efl_Config.h"`

But the file `Efl_Config.h` is provided by efl. It is thus in an unrelated directory.

Maybe the following should help:

 1. use angle brackets <>;
 2. add `-I` directives with path to efl include dir to the cflags in `elementary*.pc`;
 3. make efl a propagated build input of elementary.

**Comment by ts468** [comment]
@vbgl, thank you very much for the detailed review! I've added EFL as a propagated build input to Elementary, so Efl_Config.h should be available now.

**Comment by vbgl** [comment]
Cool. But not enough, if I’m not mistaken. Following instructions from http://docs.enlightenment.org/auto/elementary/group__Start.html yields an error when running `make`:

~~~
cc -DHAVE_CONFIG_H -I. -I.   -I/nix/store/xyiiynmkbf8v0mlcccsc13zzkzmrqdhg-elementary-1.12.3/include/elementary-1  -g -O2 -MT myapp-main.o -MD -MP -MF .deps/myapp-main.Tpo -c -o myapp-main.o `test -f 'main.c' || echo './'`main.c
In file included from main.c:1:0:
/nix/store/xyiiynmkbf8v0mlcccsc13zzkzmrqdhg-elementary-1.12.3/include/elementary-1/Elementary.h:66:24: fatal error: Efl_Config.h: No such file or directory
 #include "Efl_Config.h"
~~~

**Comment by ts468** [comment]
@vbgl, thanks, but could you maybe explain why that error surprises you? How should cc know about the location of the headers of the efl library? Are you using that command in a nix-shell environment or similar? Otherwise I would expect that you would have to either manually specify all include paths or use something like pkgconfig.

**Comment by vbgl** [comment]
The error message is unexpected because the official documentation (linked above) says it should work. The command shown above has been generated using pkg-config. My point is also about pkg-config file being broken.

**Comment by ts468** [comment]
Thanks. What happens if you use pkgconfigUpstream, so the vanilla version of pkgconfig? It might be that new, unmet dependencies show up now for pgk-config, though.

File: pkgs/desktops/e19/elementary.nix [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
   src = fetchurl {
     url = "http://download.enlightenment.org/rel/libs/elementary/${name}.tar.gz";
-    sha256 = "0qjbhdhnw6h5zc05r2w2j38yapcv9km92k5spcw3zsdnc784vf5i";
+    sha256 = "0ajljrr72rdz9hniqvsigw7p6dmpqnf0nijrxd2x91gx2cfrwksm";
   };
-  buildInputs = [ pkgconfig e19.efl gdbm ] ++ stdenv.lib.optionals stdenv.isLinux [ libcap ];
+  buildInputs = [ pkgconfig gdbm ] ++ stdenv.lib.optionals stdenv.isLinux [ libcap ];
+  propagatedBuildInputs = [ e19.efl ];
   preConfigure = ''
     export NIX_CFLAGS_COMPILE="-I${e19.efl}/include/ethumb-1 $NIX_CFLAGS_COMPILE"
```
Comment (matejc) [inline pkgs/desktops/e19/elementary.nix]:
```
export NIX_CFLAGS_COMPILE="-I${e19.efl}/include/ethumb-1 -I${e19.efl}/include/efl-1 $NIX_CFLAGS_COMPILE"
```

**Comment by matejc** [comment]
`Efl_Config.h` is in ${e19.efl}/include/ and is quite a normal fix:
```
export NIX_CFLAGS_COMPILE="-I${e19.efl}/include/ethumb-1 -I${e19.efl}/include/efl-1 $NIX_CFLAGS_COMPILE"
```

**Comment by matejc** [comment]
and do not put the `e19.efl` in the `propagatedBuildInputs`

**Comment by matejc** [comment]
and @ts468 thanks for taking care of this :beers:

**Comment by ts468** [comment]
@matejc, I've updated the PR accordingly. But if I understood @vbgl 's problem correctly, then he couldn't build an example because pkg-config did not provide all dependencies. That isn't fixed with the changes though. To my understanding it would require to rewrite the pkg-config definitions of efl/elementary to work with the nixos-special-non-common-patched version of pkg-config.

Well, but for now I think you could just merge it.

**Comment by matejc** [comment]
great, but I am afraid to merge it :P
It works on unstable branch, but I did not test it on master.
I tried with `nox-review pr 6003` but it fails with many merge conflicts.. (its probably not your fault)
Could you rebase to master?

**Comment by ts468** [comment]
Done.

**Comment by vcunat** [comment]
Still builds OK.

**Pull request closed without merging**